src/Command/SplitDb.php

```diff
@@ -0,0 +1,269 @@
+    {
+        try {
+            $types = $input->getArgument(self::ARGUMENT_TYPES);
+            $configFormEnvFile = $this->configReader->read() ?? [];
+            $envDbConfig = $this->getDatabaseConfig();
+
+            $breakExecution = false;
+            foreach ($types as $type) {
+                $connectionName = self::TYPE_MAP[$type][DbConfig::KEY_CONNECTION];
+                if (isset($configFormEnvFile['db']['connection'][$connectionName])) {
+                    $output->writeln(sprintf('Database for %s has been split.', $type));
+                    $breakExecution = true;
+                    continue;
+                }
+                if (!isset($envDbConfig['connection'][$connectionName])) {
+                    $output->writeln(sprintf('ERROR: there is not connections to additional DBs for %s splitting.', $type));
+                    $breakExecution = true;
+                }
+            }
+
+            if ($breakExecution) {
+                return;
+            }
+
+            if (!$this->maintenanceMode->isEnabled()) {
+                $helper = $this->getHelper('question');
+                $question = new ConfirmationQuestion(
+                    'We recommend to run this command on the project in maintenance mode.' .
+                    ' For this type N and run maintenance:enable CLI command.' . PHP_EOL
+                    . 'Do you want to continue without maintenance mode [y|N]?',
+                    false
+                );
+                if (!$helper->ask($input, $output, $question) && $input->isInteractive()) {
+                    return;
+                }
+            }
+            $useSlave = $this->stageConfig->get(DeployInterface::VAR_MYSQL_USE_SLAVE_CONNECTION);
+            foreach ($types as $type) {
+                $connectionName = self::TYPE_MAP[$type][DbConfig::KEY_CONNECTION];
+                $splitDbConfig = $envDbConfig['connection'][$connectionName];
+                $cmd = sprintf(
+                    'setup:db-schema:split-%s --host="%s" --dbname="%s" --username="%s" --password="%s"',
+                    $type,
+                    $splitDbConfig['host'],
+                    $splitDbConfig['dbname'],
+                    $splitDbConfig['username'],
+                    $splitDbConfig['password']
+                );
+                $outputCmd = $this->magentoShell->execute($cmd)->getOutput();
+                $output->writeln('CMD output: ' . $outputCmd);
+                $output->writeln(sprintf('Quote tables were split to DB %s in %s', $splitDbConfig['dbname'], $splitDbConfig['host']));
+
+                if ($useSlave) {
+                    $splitDbConfigSlave = $envDbConfig['slave_connection'][$connectionName];
+                    $resourceName = self::TYPE_MAP[$type][DbConfig::KEY_RESOURCE];
+                    $cmd = sprintf('setup:db-schema:add-slave --host="%s" --dbname="%s" --username="%s" --password="%s" --connection="%s" --resource=%s',
```

Comment:
Should not we check Magento version to be sure that it does support this `add-slave` command?